iamclient/iamclient.go

```diff
@@ -273,6 +272,62 @@ func (client *Client) Close() (err error) {
  * Private
  **********************************************************************************************************************/
 
+func createPublicConnection(
+	serverURL string, cryptocontext *cryptutils.CryptoContext,
+	insecure bool) (connection *grpc.ClientConn, err error) {
+	var secureOpt grpc.DialOption
+
+	ctx, cancel := context.WithTimeout(context.Background(), iamRequestTimeout)
+	defer cancel()
+
+	if insecure {
+		secureOpt = grpc.WithInsecure()
+	} else {
+		tlsConfig, err := cryptocontext.GetClientTLSConfig()
+		if err != nil {
+			return nil, aoserrors.Wrap(err)
+		}
+
+		secureOpt = grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig))
+	}
+
+	if connection, err = grpc.DialContext(ctx, serverURL, secureOpt, grpc.WithBlock()); err != nil {
+		return nil, aoserrors.Wrap(err)
+	}
+
+	return connection, nil
+}
+
+func (client *Client) createProtectedConnection(
+	config *config.Config, cryptocontext *cryptutils.CryptoContext,
+	insecure bool) (connection *grpc.ClientConn, err error) {
+	var secureOpt grpc.DialOption
+
+	ctx, cancel := context.WithTimeout(context.Background(), iamRequestTimeout)
```

Comment:
please create context before use () move to line 323

Reply:
Done